ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/use-fees-calculations.ts

```diff
@@ -0,0 +1,167 @@
+import { EtherDenomination } from '../../../../../../../../shared/constants/common';
+import {
+  addHexes,
+  getEthConversionFromWeiHex,
+  getValueFromWeiHex,
+  multiplyHexes,
+} from '../../../../../../../../shared/modules/conversion.utils';
+import { getConversionRate } from '../../../../../../../ducks/metamask/metamask';
+import { useFiatFormatter } from '../../../../../../../hooks/useFiatFormatter';
+import { getCurrentCurrency } from '../../../../../../../selectors';
+
+export function useFeeCalculations(
+  hasLayer1GasFee: boolean,
+  layer1GasFee: string,
+  gasEstimate: string,
+  supportsEIP1559: boolean,
+  transactionMeta: TransactionMeta,
+  maxPriorityFeePerGas: string,
+  maxFeePerGas: string,
+) {
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const conversionRate = useSelector(getConversionRate);
+  const fiatFormatter = useFiatFormatter();
+
+  const getFeesFromHex = (hexFee: string) => {
+    const nativeCurrencyFee = getEthConversionFromWeiHex({
+      value: hexFee,
+      fromCurrency: EtherDenomination.GWEI,
+      numberOfDecimals: 4,
+    });
+
+    const currentCurrencyFee = fiatFormatter(
+      Number(
+        getValueFromWeiHex({
+          value: hexFee,
+          conversionRate,
+          fromCurrency: EtherDenomination.GWEI,
+          toCurrency: currentCurrency,
+          numberOfDecimals: 2,
+        }),
+      ),
+    );
+
+    return { currentCurrencyFee, nativeCurrencyFee };
+  };
+
+  const defaultFee = '';
+  const [
+    currentCurrencyEstimatedFeeFromBreakdown,
+    setCurrentCurrencyEstimatedFeeFromBreakdown,
+  ] = useState(defaultFee);
+  const [
+    nativeCurrencyEstimatedFeeFromBreakdown,
+    setNativeCurrencyEstimatedFeeFromBreakdown,
+  ] = useState(defaultFee);
+  const [currentCurrencyEstimatedFee, setCurrentCurrencyEstimatedFee] =
+    useState(defaultFee);
+  const [nativeCurrencyEstimatedFee, setNativeCurrencyEstimatedFee] =
+    useState(defaultFee);
+  const [currentCurrencyL1Fee, setCurrentCurrencyL1Fee] = useState(defaultFee);
+  const [nativeCurrencyL1Fee, setNativeCurrencyL1Fee] = useState(defaultFee);
+  const [currentCurrencyL2Fee, setCurrentCurrencyL2Fee] = useState(defaultFee);
+  const [nativeCurrencyL2Fee, setNativeCurrencyL2Fee] = useState(defaultFee);
+
+  useEffect(() => {
+    if (hasLayer1GasFee) {
+      // L1
+      const feesL1 = getFeesFromHex(layer1GasFee);
+      setCurrentCurrencyL1Fee(feesL1.currentCurrencyFee);
+      setNativeCurrencyL1Fee(feesL1.nativeCurrencyFee || defaultFee);
+
+      // L2
+      const feesL2 = getFeesFromHex(gasEstimate);
+      setCurrentCurrencyL2Fee(feesL2.currentCurrencyFee);
+      setNativeCurrencyL2Fee(feesL2.nativeCurrencyFee || defaultFee);
+
+      // Estimated fee from breakdown = L1 + L2
+      const estimatedFeeFromBreakdown = addHexes(gasEstimate, layer1GasFee);
+      const feesFromBreakdown = getFeesFromHex(estimatedFeeFromBreakdown);
+      setCurrentCurrencyEstimatedFeeFromBreakdown(
```

Comment:
Assuming none of these values are from async calls, would these two `useEffect` usages be a great use case for a `useMemo` since we're just caching the calculations and not syncing with an external system or recording user input?

It may help readability alot as we can directly assign to variables rather than relying on set state methods.